e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/sort-premium-payments.spec.js

```diff
@@ -0,0 +1,117 @@
+
+      // Facility 22222222 has status 'DOES_NOT_MATCH' (alphabetically first) and has the lowest total reported payments
+      // (secondary sorting criteria, ascending), so it should appear first.
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(0).first('th').contains('22222222');
+
+      // Facility 11111111 also has status 'DOES_NOT_MATCH' but has a slightly higher total reported payments, so it
+      // should appear second.
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(1).first('th').contains('11111111');
+
+      // Facility 33333333 has status 'TO_DO' (alphabetically last), so it should appear last..
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(2).first('th').contains('33333333');
+    });
+  });
+
+  describe('when total payment received column heading is clicked', () => {
+    it('should sort the rows by total payment received', () => {
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.header.totalPaymentsReceived().click();
+
+      // Check that the rows are sorted by total payment received by checking the facility IDs.
+
+      // Facility 33333333 has no reported payments, so it should appear first.
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(0).first('th').contains('33333333');
+
+      // Facility 22222222 has the highest total reported payments (200 EUR), so it should appear second.
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(1).first('th').contains('22222222');
+
+      // Facility 11111111 has the lowest total reported payments (300 USD), so it should appear last.
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(2).first('th').contains('11111111');
+    });
+  });
+
+  describe('when total reported payments column heading is clicked', () => {
+    it('should sort the rows by total reported payments', () => {
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.header.totalReportedPayments().click();
+
+      // Check that the rows are sorted by total reported payments by checking the facility IDs.
+
+      // NOTE: This column is the default sorting column and defaults to ascending, therefore the descending case is being tested.
+
+      // Facility 11111111 has the highest total reported payments (30 EUR), so it should appear first.
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(0).first('th').contains('11111111');
+
+      // Facility 22222222 has the second highest total reported payments (20 GBP), so it should appear second.
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(1).first('th').contains('22222222');
+
+      // Facility 33333333 has the lowest total reported payments (10 USD), so it should appear last.
```

Comment:
I actually think these are in this order because first we order by the currency alphabetically? (e.g. 1000 USD would still come last even though actually highest)

Reply:
Yes thanks, had my currencies mixed up missed that, pushed updates for this